### PR TITLE
Assemble and organise devfile integration tests Files

### PR DESCRIPTION
**Pull request by prietyc123**
**What type of PR is this?**

/kind design

**What does does this PR do / why we need it**:

Assemble all the devfile integration tests Files into tests/integration/devfile

**Which issue(s) this PR fixes**:

Fixes NA

**How to test changes / Special notes to the reviewer**:

Run ```make test-integration-devfile```

### CODE REVIEW

**Comment by amitkrout** [comment]
Thanks @prietyc123 for the pr.

@johnmcollier @GeekArthur @yangcao77 @kanchwala-yusuf Lets add devfile specific test in a separate [package](https://github.com/openshift/odo/tree/master/tests/integration/devfile) as later it will be easy to understand and maintain specific test if the volume of devfile test contribution increases.

Thanks for understanding

File: tests/integration/devfile/cmd_devfile_catalog_test.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package integration
+package devfile
```
Comment (amitkrout) [inline tests/integration/devfile/cmd_devfile_catalog_test.go]: /lgtm

**Comment by GeekArthur** [comment]
We may need someone adds `approved` label to trigger openshift bot for helping us retest the PR automatically.

**Comment by johnmcollier** [comment]
> @johnmcollier @GeekArthur @yangcao77 @kanchwala-yusuf Lets add devfile specific test in a separate package as later it will be easy to understand and maintain specific test if the volume of devfile test contribution increases.

@amitkrout That was actually a typo on my end, I always intended for the devfile tests to be in a separate package when I moved them under `tests/integration/devfile` 🙂

**Comment by amitkrout** [comment]
> We may need someone adds `approved` label to trigger openshift bot for helping us retest the PR automatically.

@girishramnani Can you pease add the approve label here, so that the pr will run continuously until it passes before auto merge. You can feel free to apply the approve label as only files has been moved to separate package without any code changes.

**Comment by amitkrout** [comment]
prow failure
/test v4.3-integration-e2e-benchmark

**Comment by amitkrout** [comment]
prow failure
/test v4.3-integration-e2e-benchmark

**Comment by amitkrout** [comment]
prow failure
/test v4.3-integration-e2e-benchmark

**Comment by amitkrout** [comment]
prow failure
/test v4.3-integration-e2e-benchmark

**Pull request merged**